src/NHSD.BuyingCatalogue.Organisations.Api/Services/ICreateOrganisationService.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Identity.Common.Results;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    public interface ICreateOrganisationService
+    {
+        Task<Result<string>> CreateAsync(CreateOrganisationRequest request);
```

Comment:
Can we return `Result<Guid>` here?

Reply:
Interesting point, thank you for raising it.

I'd prefer the service to return a string representing the guid - as that's what gets propagated to the (wrongly named) viewModel.

Why doesnt the 'viewModel' return the Guid rather than string?
 I chose the current implementation because it felt right. I know that's not a strong argument, however I felt wrong about returning a Guid without having a concrete understanding of what/where and how serialises that Guid to JSON (which is not a concern with a string).

I had a look and the rework to return a Guid all the way through is not huge (6 files).

Conclusion, I'm not a fan, however, if you shed some more light on why should we return a Guid instead of a string I'm willing to compromise and change it.

Reply:
`CreateOrganisationResponseViewModel.OrganisationId` can be a `Guid` – it will be serialized as a string. Sorry, I probably just have picked that up and left a comment.

The reason to prefer a `Guid` is type-safety and accurate object modelling. In the OO world we often have a tendency to be a bit lazy with our typing, such as using `string` for URLs when we should be using `Uri`. There are of course plenty of other examples too, such as using `string` for e-mails etc. – no doubt you'll be able to think of many more examples. Functional programmers tend to be much better at defining specific types... 

When we go for the `string` option we lose the advantages that the strong-typing brings – such as making sure we always have a valid Guid as our ID. It also means we can take advantage of any useful operations that the Guid struct provides (although we may not need them right now).

Therefore, I would still recommend refactoring to use the `Guid` type – particularly as it's already available in the framework.

P.S. I wasn't at all suggesting that you've been lazy in your implementation – anything but as you've obviously given it some thought! 👍

Reply:
Alright I'm sold. Guid it is 😄